src/clj/api/server.clj

```diff
@@ -25,6 +25,50 @@
 
 (declare server)
 
+;; NOTE: this is not ideal as every copy of the ECS service will maintain it's own copy
+;; It would be best to persist it to the RDB
+;; however as we are currently running only one task per service it will suffice
+(def jailed-ips (atom {}))
+
+(defn update-ip [ip timestamp]
+  (swap! jailed-ips update-in [ip :timestamps] conj timestamp))
+
+(defn is-jailed? [ip now]
```

Comment:
I find the names a little confusing, I would choose something more descriptive like `check-and-maybe-jail` for this one and maybe write the condition more like :
```clojure
(when (and info
      ;; 5 times
      (> (count timestamps) 4)
      ;; within 10 minutes
      (< (- now (last timestamps)) (* 10 60 1000))
      (not (:jail-lift-time info)))
  ;; jailed for 20 minutes
  (swap! jailed-ips update-in [ip :jail-lift-time] (fn [_] (+ now (* 20 60 1000)))))))
```

Reply:
resolving as outdated